StripeFinancialConnections/StripeFinancialConnections/Source/Native/Shared/NetworkingOTPView/NetworkingOTPDataSource.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  NetworkingOTPDataSource.swift
+//  StripeFinancialConnections
+//
+//  Created by Krisjanis Gaidis on 2/28/23.
+//
+
+import Foundation
+@_spi(STP) import StripeCore
+
+protocol NetworkingOTPDataSourceDelegate: AnyObject {
+    func networkingOTPDataSource(_ dataSource: NetworkingOTPDataSource, didUpdateConsumerSession consumerSession: ConsumerSessionData)
+}
+
+protocol NetworkingOTPDataSource: AnyObject {
+    var otpType: String { get }
+    var analyticsClient: FinancialConnectionsAnalyticsClient { get }
+    var pane: FinancialConnectionsSessionManifest.NextPane { get }
+
+    func lookupConsumerSession() -> Future<LookupConsumerSessionResponse>
+    func startVerificationSession() -> Future<ConsumerSessionResponse>
+    func confirmVerificationSession(otpCode: String) -> Future<ConsumerSessionResponse>
+}
+
+final class NetworkingOTPDataSourceImplementation: NetworkingOTPDataSource {
+
+    let otpType: String
+    private let emailAddress: String
+    private let customEmailType: String?
+    private let connectionsMerchantName: String?
+    private var consumerSession: ConsumerSessionData? {
+        didSet {
+            if let consumerSession = consumerSession {
```

Comment:
ah ok maybe the session override is just handled here vs in 3 places. Makes sense